### ISSUE TITLE
Flags  "delete-local-data" & "ignore-daemonsets"

### ISSUE THREAD

**Issue by rstreics** [aws/aws-node-termination-handler#20]
Could please add ability to configure this flags 
* --ignore-daemonsets=false: Ignore DaemonSet-managed pods
* --delete-local-data=false: Continue even if there are pods using emptyDir (local data that will be deleted when the node is drained)

**Comment by bwagner5** [comment]
Yeah, I think those two args to drain make sense to be configurable. 

I'm curious why you would want `--delete-local-data=false`. Is the idea that you want anything using emptyDir to run as long as possible even though they are about to receive a SIGKILL? That's the idea on `--ignore-daemonsets=true` (our default), so that the node can continue to operate while the other pods are gracefully shutting down. Just curious on the `delete-local-data=false` use-case for spot interrupts.

**Comment by rstreics** [comment]
Hi, for my personal use-case I need to to `delete-local-data=true` & `ignore-daemonsets=false`.
 I specified feature suggestion defaults from `kubectl`. Regarding spot-instances this `delete-local-data=true` & `ignore-daemonsets=false`default settings, because nodes will shutdown anyway. 
I assumed those flag could be configurable.

**Comment by kivagant-ba** [comment]
Hello. I'm also trying to use the node-termination-handler. That's what I got in logs:

```
2019/11/15 07:18:56 cannot delete Pods with local storage (use --delete-local-data to override): ...//a long list of pods//...
```

Am I understand correctly and it does not really helps to drain pods in my case?

**Comment by rstreics** [comment]
@kivagant-ba in short this feature request would solve this. If you are using `emptyDir` this tool in current release won't drain the nodes.

**Comment by bwagner5** [comment]
Yep, hoping to have a PR in this week.

**Comment by bwagner5** [comment]
I was slightly late on the PR, but it's open now ^^

**Comment by bwagner5** [comment]
Also @kivagant-ba and @rstreics I have defaulted `delete-local-data` to `true`. I think that makes the most sense for spot interruptions, since the whole node is about to go down anyways. You could make the case that you're using an EBS volume that can persist but I think the spirit of emptyDir is that it lives with the node and you should be using a Persistent Volume if you want to persist across nodes. But let me know if you disagree with that default.

**Comment by kivagant-ba** [comment]
Thank you, @bwagner5! I will test the change later.

**Comment by kivagant-ba** [comment]
@bwagner5 Are there any newer images somewhere? I'd like to test the changes but seems like they have not been released yet.

**Comment by bwagner5** [comment]
The latest image is published as v1.0.0. You can pull this specific hash if you're caching a local version pulled in the past. `amazon/aws-node-termination-handler:v1.0.0@sha256:b8feb9e33b1dd02961496eb32f473cedd8d33cfd8741a39c412783d603fa30de`

v1.0.0 is now pinned at the github release, so this will never change now.

I tested below to make sure configuration options are in that image:

```
docker run -it -e DRY_RUN=true -e NODE_NAME=test amazon/aws-node-termination-handler:v1.0.0@sha256:b8feb9e33b1dd02961496eb32f473cedd8d33cfd8741a39c412783d603fa30de
aws-node-termination-handler arguments:
	dry-run: true,
	node-name: test,
	metadata-url: http://169.254.169.254,
	kubernetes-service-host: ,
	kubernetes-service-port: ,
	delete-local-data: true,
	ignore-daemon-sets: true
	grace-period: -1
```
